### PR TITLE
chore: refactor to stop using the cache

### PR DESCRIPTION
**Pull request by renaudjester**
Fix: [CMT-2](https://cms-change.atlassian.net/browse/CMT-2) and stop using the cache

- [ ] Dataset - product mapping is ready

### CODE REVIEW

File: copernicusmarine/core_functions/subset.py [hunk 1]
```diff
@@ -99,17 +85,17 @@ def subset_function(
...
-        subset_request = subset_request_from_file(request_file)
+        subset_request.from_file(request_file)
+    if not subset_request.dataset_id:
+        raise ValueError("Please provide a dataset id for a subset request.")
     if motu_api_request:
         motu_api_subset_request = convert_motu_api_request_to_structure(
             motu_api_request
         )
         subset_request.update(motu_api_subset_request.__dict__)
     request_update_dict = {
```
Comment (uriii3) [inline copernicusmarine/core_functions/subset.py:98]: are we sure that you can delete the update of the id? nothing weird with python variables?

File: copernicusmarine/core_functions/subset.py [hunk 2]
```diff
@@ -153,30 +138,10 @@ def subset_function(
...
-                )
-            else:
-                syntax_error = SyntaxError(
-                    "Must specify at least one of "
-                    "'dataset_url' or 'dataset_id' options"
-                )
-                raise syntax_error
-        else:
-            dataset_id = subset_request.dataset_id
+            raise SyntaxError("Must specify 'dataset_id' option")
```
Comment (renaudjester) [inline copernicusmarine/core_functions/subset.py:141]: To delete

File: copernicusmarine/core_functions/subset.py (hunk 1 above)
Comment (renaudjester) [reply copernicusmarine/core_functions/subset.py:98]: I don't think so but better be on the side of caution

Comment (renaudjester) [reply copernicusmarine/core_functions/subset.py:96]: I checked and it seems to work with the obvious cases

**Review by uriii3 — APPROVED** [review]

**Pull request merged**